torch/_inductor/shape_propagation.py

```diff
@@ -83,9 +83,11 @@ class ShapePropagationOpsHandler:
     @staticmethod
     def constant(value: torch.types.Number, dtype: torch.dtype) -> BlockShapeType:
         # See implementation of constant for triton for the reason
```

Comment:
The implementation https://github.com/pytorch/pytorch/blob/d3bdf8c32ea5f14a3c803c45f9a9489bfba1d9ca/torch/_inductor/codegen/triton.py#L1033-L1053 has a special code path for `tl.float32`. I'm matching it here.